pkg/eventsourcing/storage/store_posgres_config.go

```diff
@@ -64,23 +66,19 @@ func NewPostgresStoreConfig(url string) (*postgresStoreConfig, error) {
 
 // ConfigureTLS adds the configuration for TLS secured connection/auth
 func (conf *postgresStoreConfig) ConfigureTLS() error {
-	cfg := &tls.Config{
-		RootCAs:    x509.NewCertPool(),
-		ServerName: strings.Split(conf.pgOptions.Addr, ":")[0],
-	}
-	if ca, err := ioutil.ReadFile("/etc/eventstore/certs/db/ca.crt"); err != nil {
+	loader, err := m8tls.NewTLSConfigLoader(CACertPath, TLSCertPath, TLSKeyPath)
+	if err != nil {
 		return err
-	} else {
-		cfg.RootCAs.AppendCertsFromPEM(ca)
 	}
 
-	if cert, err := tls.LoadX509KeyPair("/etc/eventstore/certs/db/tls.crt", "/etc/eventstore/certs/db/tls.key"); err != nil {
+	err = loader.Watch()
```

Comment:
where exactly does the internal watch routine come into effect?  In this code the loaded certs are assigned to a postgres tls config and then it's returned. How is it refreshed?

Reply:
clarified verbally. since tls_config is returning functions to the postgres config this is automatically renewed